FoundationDB/fdb-kubernetes-operator
# Using explicit listen address for all clusters

# Description

This adds an option to use an explicit listen address in the monitor conf regardless of the IP source.

Resolves #393

## Type of change

*Please select one of the options below.*

- New feature (non-breaking change which adds functionality)

# Discussion

*Are there any design details that you would like to discuss further?*

I tried different rollout strategies and decided that a rolling bounce was the best way to go. This means that we're tracking more information in the status, so we know when it's safe to update the monitor conf.

# Testing

*Please describe the tests that you ran to verify your changes. Unit tests?
Manual testing?*

I created a cluster in my dev environment and cycled it through enabling and disabling the new listen address flag. I also cycled through the pod/service options for the IP source to make sure this still works.

*Do we need to perform additional testing once this is merged, or perform in a larger testing environment?*

We may want to add this to the integration tests, since there are subtle aspects to how the environment variables and the monitor conf get rolled out.

# Documentation

*Did you update relevant documentation within this repository?*

Yes

*If this change is adding new functionality, do we need to describe it in our user manual?*

No, I don't think so

*If this change is adding or removing subreconcilers, have we updated the core technical design doc to reflect that?*

N/A

*If this change is adding new safety checks or new potential failure modes, have we documented and how to debug potential issues?*

N/A

# Follow-up

*Are there any follow-up issues that we should pursue in the future?*

Yes, once the change to the defaults gets made in the next release, we should deprecate this option and the entire behavior of implicit listen addresses. We should also remove support for binding to FDB_PUBLIC_IP in the sidecar, and drop the behavior of passing FDB_POD_IP as an explicit substitution to the sidecar since newer versions of the sidecar support it without any flag.

*Does this introduce new defaults that we should re-evaluate in the future?*

It introduces new defaults, but they are set to change in the next release.
